### PR TITLE
PB-12 Recover password flow

### PR DESCRIPTION
**Pull request by igoscheff**

### CODE REVIEW

File: phonebook-ui/src/app/common/service/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { Observable } from "rxjs";
+import { User } from "../interface/user.interface";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
```
Comment (sharf85) [inline phonebook-ui/src/app/common/service/auth.service.ts:9]: Use UserService for working with user

File: phonebook-ui/src/app/auth/forgot-password-page/forgot-password-page.component.css [hunk 2]
```diff
@@ -0,0 +1,151 @@
+.auth-form {
```
Comment (sharf85) [inline phonebook-ui/src/app/auth/forgot-password-page/forgot-password-page.component.css:1]: what is this? Is it source or result? In git there should be only sources and instruction, if needed, how to transpile them into end files

File: phonebook-ui/src/app/common/service/user.service.ts [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
 
 @Injectable({
   providedIn: 'root'
 })
 export class UserService {
 
-  private host = 'http://localhost:8080/';
-  private registerUrl = `${this.host}api/v1/user/registration`;
-  private registerConfirmUrl = `${this.host}api/v1/user/confirmation?token=`;
+  private host = 'http://localhost:8080/'
```
Comment (sharf85) [inline phonebook-ui/src/app/common/service/user.service.ts:11]: if we use proxi.conf.json , we dont need this var

Comment (igoscheff) [reply phonebook-ui/src/app/common/service/user.service.ts:11]: It's not my code, I didn't touch it!

Comment (sharf85) [reply phonebook-ui/src/app/common/service/user.service.ts:11]: And what?=))

File: phonebook-ui/src/app/common/service/user.service.ts [hunk 4]
```diff
@@ -1,24 +1,30 @@
...
   sendRequestToRegisterConfirm(token: string) {
-    return this.http.get(`${this.registerConfirmUrl}${token}`);
+    return this.http.get(`/api/user/confirmation?token=${token}`)
+  }
+
+  recoveryPassword(user: User): Observable<User> {
+    return this.http.post<User>('/api/reset-password', user)
   }
+
+  resetPassword(user: User, token: string): Observable<User> {
```
Comment (sharf85) [inline phonebook-ui/src/app/common/service/user.service.ts:26]: great . And please no string expressions in the code - make constants either in the class or in the file below the class.

Comment (igoscheff) [reply phonebook-ui/src/app/common/service/user.service.ts:26]: where are string expressions?

**Pull request merged**